### ISSUE TITLE
Renaming of function, when done twice, loses track of change

### ISSUE THREAD

**Issue by Kaspazza** [clojure-lsp/clojure-lsp#1795]
**Describe the bug**
This bug didn't existed in 05.03 release. 
The problem is that when renaming the same function twice, lsp loses track of it's length, making second rename not applied or applied incorrectly (e.g. mixing name with part of it's variables)

**To Reproduce**
1. Go to function-name that is used in other namespaces
2. Execute lsp-rename and change name to longer or shorter name
3. Save the change - everything looks good but highlighting is lost 
4. Rename the same function again, to an even shorter (or longer name)
5. The name is messed up or rename was not showing proper name

**Expected behavior**
When I rename a function I want the name to be refreshed in all it's references. 
So if I rename it to a longer name, I want it in another namespace where it's referenced to be able to rename it from there again. 
If I rename a function from the definition point, I want to be able to do it twice there. 

**Screenshots**
[Video recording of the issue
](https://youtu.be/sX9qcM5OiNg)**User details (please complete the following information):**

 - OS: MacOS aarch binary
 - Editor: doom emacs
 - Version: 2024.03.17-17.22.15-nightly

**Additional context**
Seems like it's the same bug coming again: https://github.com/clojure-lsp/clojure-lsp/issues/1755

**Comment by Kaspazza** [comment]
Happens also in [2024.03.13-13.11.00](https://github.com/clojure-lsp/clojure-lsp/releases/tag/2024.03.13-13.11.00)

**Comment by Kaspazza** [comment]
Happens also in https://github.com/clojure-lsp/clojure-lsp/releases/tag/2024.03.01-11.37.51

**Comment by Kaspazza** [comment]
Also in [2024.02.01-11.01.59](https://github.com/clojure-lsp/clojure-lsp/releases/tag/2024.02.01-11.01.59)

**Comment by julienvincent** [comment]
I get this as well - basically makes using rename unusable. Any change this is something that could get looked at?

**Comment by ericdallo** [comment]
Yeah, save all files is the best workaround for now, for emacs this works great:
```lisp
(add-hook 'lsp-after-apply-edits-hook (lambda (&rest _) (save-buffer)))
```

**Comment by julienvincent** [comment]
@Kaspazza @ericdallo I am working in neovim and saving all buffers does not solve the issue. The only thing I have found to resolve once it gets into a broken state is to fully restart neovim, which is obviously not a good solution.

> so not sure it's a clojure-lsp bug

I've used similar rename and refactoring code actions from other language servers such as rust-analyser and tsserver without ever running into this issue. Maybe I'm wrong but I'm fairly sure this is a clojure-lsp specific bug.

**Comment by blueskyonmars** [comment]
Are you still able to reproduce this?  

I'm using VSCode/Calva on Fedora 44 with clojure-lsp:
clojure-lsp 2026.05.05-12.58.26  (actually this is a build based on git commit 194829088c7a82005aab2fedcd5d54a7af519e99)
clj-kondo 2026.05.26-SNAPSHOT

and I can't get it to reproduce.
